data_diff/__main__.py

```diff
@@ -402,6 +409,16 @@ def _data_diff(
         )
         return
 
+    if init_oracle_client_lib_dir:
```

Comment:
I think that we should move this oracle related code to https://github.com/datafold/data-diff/blob/master/data_diff/sqeleton/databases/oracle.py

And it could be enabled by an environment variable instead. e.g. `LEGACY_ORACLE_LIB`

The main method here is already messy and don't want to make it worse